gobblin-service/src/main/java/org/apache/gobblin/service/modules/orchestration/DagManager.java

```diff
@@ -853,27 +852,40 @@ private boolean killJobIfOrphaned(DagNode<JobExecutionPlan> node, JobStatus jobS
             DagManagerUtils.getFullyQualifiedDagName(node),
             timeOutForJobStart);
         dagManagerMetrics.incrementCountsStartSlaExceeded(node);
-        cancelDagNode(node);
 
         String dagId = DagManagerUtils.generateDagId(node).toString();
-        this.dags.get(dagId).setFlowEvent(TimingEvent.FlowTimings.FLOW_START_DEADLINE_EXCEEDED);
+        this.dags.get(dagId).setFlowEvent(TimingEvent.FlowTimings.FLOW_CANCELLED);
```

Comment:
This one seems like a discrepancy, if we want to retry because job start sla exceeded, we will set dagNode with pending_retry but flow status to be flow_cancelled?

Reply:
we're not calling `cancelDagNode` but still updating flow status also